### ISSUE TITLE
hit_collection.rb:63:in `select': undefined method `select!' for nil:NilClass (NoMethodError)

### ISSUE THREAD

**Issue by CAMOBAP** [relaton/relaton-bib#46]
## Stacktrace

<details><summary>Full stack trace</summary>
<p>

```
/Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/relaton-bib-1.7.5/lib/relaton_bib/hit_collection.rb:63:in `select': undefined method `select!' for nil:NilClass (NoMethodError)
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/relaton-iso-1.7.5/lib/relaton_iso/iso_bibliography.rb:126:in `search_code'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/relaton-iso-1.7.5/lib/relaton_iso/iso_bibliography.rb:81:in `isobib_search_filter'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/relaton-iso-1.7.5/lib/relaton_iso/iso_bibliography.rb:177:in `isobib_get1'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/relaton-iso-1.7.5/lib/relaton_iso/iso_bibliography.rb:37:in `get'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/relaton-iso-1.7.5/lib/relaton_iso/processor.rb:19:in `get'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/relaton-1.7.9/lib/relaton/db.rb:420:in `net_retry'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/relaton-1.7.9/lib/relaton/db.rb:405:in `new_bib_entry'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/relaton-1.7.9/lib/relaton/db.rb:385:in `check_bibliocache'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/relaton-1.7.9/lib/relaton/db.rb:73:in `fetch'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/ref_sect.rb:70:in `fetch_ref'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-nist-1.3.1/lib/asciidoctor/nist/converter.rb:213:in `fetch_ref'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/ref.rb:162:in `refitem1'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/ref.rb:154:in `refitem'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-nist-1.3.1/lib/asciidoctor/nist/converter.rb:208:in `refitem'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/ref.rb:213:in `reference1'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/ref_sect.rb:14:in `block (2 levels) in reference'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/ref_sect.rb:14:in `each'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/ref_sect.rb:14:in `block in reference'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:295:in `initialize'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:254:in `new'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:254:in `with'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/utils.rb:35:in `noko'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/ref_sect.rb:13:in `reference'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/lists.rb:39:in `ulist'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/utils.rb:14:in `convert'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/asciidoctor-2.0.15/lib/asciidoctor/abstract_block.rb:75:in `convert'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/asciidoctor-2.0.15/lib/asciidoctor/list.rb:30:in `convert'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/asciidoctor-2.0.15/lib/asciidoctor/abstract_block.rb:84:in `block in content'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/asciidoctor-2.0.15/lib/asciidoctor/abstract_block.rb:84:in `map'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/asciidoctor-2.0.15/lib/asciidoctor/abstract_block.rb:84:in `content'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-nist-1.3.1/lib/asciidoctor/nist/converter.rb:184:in `block (2 levels) in bibliography_annex_parse'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:394:in `insert'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:377:in `method_missing'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-nist-1.3.1/lib/asciidoctor/nist/converter.rb:183:in `block in bibliography_annex_parse'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:394:in `insert'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:377:in `method_missing'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-nist-1.3.1/lib/asciidoctor/nist/converter.rb:180:in `bibliography_annex_parse'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-nist-1.3.1/lib/asciidoctor/nist/converter.rb:174:in `bibliography_parse'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/section.rb:79:in `block in section'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:295:in `initialize'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:254:in `new'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:254:in `with'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/utils.rb:35:in `noko'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/section.rb:64:in `section'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/utils.rb:14:in `convert'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/asciidoctor-2.0.15/lib/asciidoctor/abstract_block.rb:75:in `convert'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/asciidoctor-2.0.15/lib/asciidoctor/abstract_block.rb:84:in `block in content'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/asciidoctor-2.0.15/lib/asciidoctor/abstract_block.rb:84:in `map'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/asciidoctor-2.0.15/lib/asciidoctor/abstract_block.rb:84:in `content'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/asciidoctor-2.0.15/lib/asciidoctor/document.rb:1013:in `content'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/base.rb:218:in `block in middle'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:394:in `insert'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:377:in `method_missing'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/base.rb:217:in `middle'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/base.rb:189:in `block in makexml1'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:295:in `initialize'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:254:in `new'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/builder.rb:254:in `with'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/utils.rb:35:in `noko'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/base.rb:189:in `makexml1'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/base.rb:195:in `makexml'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-nist-1.3.1/lib/asciidoctor/nist/converter.rb:87:in `makexml'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/base.rb:161:in `document'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-standoc-1.9.2/lib/asciidoctor/standoc/utils.rb:14:in `convert'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/asciidoctor-2.0.15/lib/asciidoctor/document.rb:944:in `convert'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/asciidoctor-2.0.15/lib/asciidoctor/convert.rb:119:in `convert'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-1.3.2/lib/metanorma/input/asciidoc.rb:24:in `process'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-1.3.2/lib/metanorma/processor.rb:24:in `input_to_isodoc'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-1.3.2/lib/metanorma/compile.rb:93:in `process_input'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-1.3.2/lib/metanorma/compile.rb:26:in `compile'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-cli-1.4.8/lib/metanorma/cli/compiler.rb:45:in `compile_file'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-cli-1.4.8/lib/metanorma/cli/compiler.rb:30:in `compile'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-cli-1.4.8/lib/metanorma/cli/compiler.rb:35:in `compile'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-cli-1.4.8/lib/metanorma/cli/site_generator.rb:81:in `compile'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-cli-1.4.8/lib/metanorma/cli/site_generator.rb:28:in `block in generate'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-cli-1.4.8/lib/metanorma/cli/site_generator.rb:28:in `each'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-cli-1.4.8/lib/metanorma/cli/site_generator.rb:28:in `generate'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-cli-1.4.8/lib/metanorma/cli/site_generator.rb:23:in `generate'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-cli-1.4.8/lib/metanorma/cli/commands/site.rb:25:in `generate'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-hollaback-0.2.0/lib/thor/hollaback.rb:62:in `run'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-1.0.1/lib/thor/invocation.rb:116:in `invoke'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-1.0.1/lib/thor.rb:243:in `block in subcommand'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-hollaback-0.2.0/lib/thor/hollaback.rb:62:in `run'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-1.0.1/lib/thor/base.rb:485:in `start'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-cli-1.4.8/lib/metanorma/cli.rb:34:in `start'
	from /Users/camobap/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/metanorma-cli-1.4.8/exe/metanorma:20:in `<top (required)>'
	from /Users/camobap/.rbenv/versions/2.4.2/bin/metanorma:23:in `load'
	from /Users/camobap/.rbenv/versions/2.4.2/bin/metanorma:23:in `<main>'
```

</p>
</details>

CI run - https://github.com/metanorma/mn-samples-nist/runs/2620948558?check_suite_focus=true#step:6:1107

## How to reproduce

Can be reproduced on the repo  https://github.com/metanorma/mn-samples-nist with command `metanorma site generate . --agree-to-terms`

As far as I understand the problem with this ref https://github.com/metanorma/mn-samples-nist/blob/master/sources/sections/800-116r1/body.adoc#L2754 , but because I'm not familiar with `relaton` I don't understand how to fix it

**Comment by andrew2net** [comment]
this issue is fixed in relaton -v 1.8.0 but I'm not sure if @opoudjis has updated metanorma to use the latest relaton version

**Comment by CAMOBAP** [comment]
I can confirm it works with 1.8.0
